plugins/com.gwtplugins.gdt.eclipse.maven/src/com/google/gdt/eclipse/maven/configurators/M2eCompat.java

```diff
@@ -0,0 +1,41 @@
+package com.google.gdt.eclipse.maven.configurators;
+
+import org.apache.maven.project.MavenProject;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
+import org.eclipse.m2e.core.project.configurator.ProjectConfigurationRequest;
+
+/**
+ * @author Luca Piccarreta
+ *
+ * Adaptation layer for m2e 1.x / 2.x API
+ */
+public class M2eCompat {
```

Comment:
This file is duplicated, but a clean solution would require an entire new module, which was a bit... overkill

Reply:
Possibly a dumb question, but wouldnt it be less overkill to do it the other way?

 * 2 java files, identical, we now have to update one when the other changes
 * 1 java file and one pom.xml, a dependency on maven core, m2e core (and the reflection util class can live there too)

Or am I missing something, and due to the "eclipse-project" nature that new module needs to have?

Reply:
It's a OSGi environment, I'm simply not sure how long it will take... I'm not the right man to do the job the way you say it.
Even though I agree that it stinks.
The other solution I can think of, is just removing this M2eCompat stuff, and inlining it, so that it is exactly the same, but stinks less

Reply:
i have some experience with plugin development. maybe i could help here.

Reply:
Could you make a PR against this branch in foosoftsrl/gwt-eclipse-plugin with the sorts of setup one would need to do this?

Alternatively, bake in to the ReflectionUtilities API a way to pass the params to build the reflected Methods (class+methodname pairs etc), and let it return a single Method/MethodHandle or throw, and then we don't need even one copy of this class at all.